dotnet/src/Connectors/Connectors.Amazon/Bedrock/Models/Amazon/AmazonIOService.cs

```diff
@@ -0,0 +1,213 @@
+    /// Builds the ConverseRequest object for the Bedrock ConverseAsync call with request parameters required by Amazon Titan.
+    /// </summary>
+    /// <param name="modelId">The model ID.</param>
+    /// <param name="chatHistory">The messages between assistant and user.</param>
+    /// <param name="settings">Optional prompt execution settings.</param>
+    /// <returns></returns>
+    public ConverseRequest GetConverseRequest(string modelId, ChatHistory chatHistory, PromptExecutionSettings settings)
+    {
+        var titanRequest = new TitanRequest.TitanChatCompletionRequest
+        {
+            Messages = chatHistory.Select(m => new Message
+            {
+                Role = MapRole(m.Role),
+                Content = new List<ContentBlock> { new() { Text = m.Content } }
+            }).ToList(),
+            System = new List<SystemContentBlock>(), // { new SystemContentBlock { Text = "You are an AI assistant." } },
+            InferenceConfig = new InferenceConfiguration
+            {
+                Temperature = this.GetExtensionDataValue(settings.ExtensionData, "temperature", 0.7f),
+                TopP = this.GetExtensionDataValue(settings.ExtensionData, "topP", 0.9f),
+                MaxTokens = this.GetExtensionDataValue(settings.ExtensionData, "maxTokenCount", 512),
+            },
+            AdditionalModelRequestFields = new Document(),
+            AdditionalModelResponseFieldPaths = new List<string>()
+        };
+        var converseRequest = new ConverseRequest
+        {
+            ModelId = modelId,
+            Messages = titanRequest.Messages,
+            System = titanRequest.System,
+            InferenceConfig = titanRequest.InferenceConfig,
+            AdditionalModelRequestFields = titanRequest.AdditionalModelRequestFields,
+            AdditionalModelResponseFieldPaths = titanRequest.AdditionalModelResponseFieldPaths,
+            GuardrailConfig = null, // Set if needed
+            ToolConfig = null // Set if needed
+        };
+        return converseRequest;
+    }
+
+    private TValue GetExtensionDataValue<TValue>(IDictionary<string, object>? extensionData, string key, TValue defaultValue)
+    {
+        if (extensionData == null || !extensionData.TryGetValue(key, out object? value))
+        {
+            return defaultValue;
+        }
+
+        if (value is TValue typedValue)
+        {
+            return typedValue;
+        }
+
+        return defaultValue;
+    }
+
+    private static ConversationRole MapRole(AuthorRole role)
+    {
+        string roleStr;
+        if (role == AuthorRole.User)
+        {
+            roleStr = "user";
+        }
+        else
+        {
+            roleStr = "assistant";
+        }
+        return roleStr switch
+        {
+            "user" => ConversationRole.User,
+            "assistant" => ConversationRole.Assistant,
+            _ => throw new ArgumentOutOfRangeException(nameof(role), $"Invalid role: {role}")
+        };
+    }
```

Comment:
This could be simplified
```
private static ConversationRole MapRole(AuthorRole role)
{
    return role switch
    {
        AuthorRole.User => ConversationRole.User,
        AuthorRole.Assistant => ConversationRole.Assistant,
        _ => throw new ArgumentOutOfRangeException(nameof(role), $"Invalid role: {role}")
    };
}

```
Also, `GetTextStreamOutput`, `MapRole` and `GetExtensionDataValue` has been defined in all models, which can be avoided by defining them in a separate utility class.

Reply:
`GetTextStreamOutput` is unique to each model but I've moved `MapRole` and `GetExtensionDataValue` into a common utility class `BedrockUtilities`